pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1249,8 +1249,8 @@ func setUpstreamProtocol(node *model.Proxy, cluster *apiv2.Cluster, port *model.
 		}
 	}
 
-	if (util.IsProtocolSniffingEnabledForInboundPort(node, port) && direction == model.TrafficDirectionInbound) ||
-		(util.IsProtocolSniffingEnabledForOutboundPort(node, port) && direction == model.TrafficDirectionOutbound) {
+	if node.Type == model.SidecarProxy && ((util.IsProtocolSniffingEnabledForInboundPort(node, port) && direction == model.TrafficDirectionInbound) ||
```

Comment:
looks good, better add some comments

Reply:
Can you please explain in comments, why enabling it for gateways is a problem so that it is more clear on why we are adding only for sidecar?

Reply:
Thanks for adding the comments, looks good